### PR TITLE
Add/Include message and then delete account

### PR DESCRIPTION
**Pull request by girishbhamare**
Account doesn't get deleted when it has a message and rest all tests gets failed due to this issue.
Now Issue got resolved.
I have added soap tests for the same.
        <t:objective>Delete a valid account which has a message</t:objective>
            <Steps> 1. Create a valid account.
                2. Auth using test account
                3. Send Msg to same account
                4. Delete the account.
        </Steps>

Result:
     [java] EXECUTING TEST: /opt/soap/zm-soap-harness/build/temp/data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml
     [java]
     [java] REQUEST/RESPONSE Results: 7(PASS) - 0(Fail)
     [java]
     [java]
     [java] Tests Executed:7
     [java] Pass:7
     [java] Fail:0
     [java]
     [java] Test Cases Executed:1
     [java] Pass:1
     [java] Fail:0
     [java] Script Errors:0

### CODE REVIEW

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                </t:response>
+            </t:test>
+        </t:test_case>
+        <t:test_case testcaseid="DeleteAccountRequest_WithMsg" type="smoke">
+            <t:objective>Delete a valid account which has a message</t:objective>
+            <Steps> 1. Create a valid account.
+                2. Auth using test account
+                3. Send Msg to same account
+                4. Delete the account.
+        </Steps>
```
Comment (chinmay15) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:74]: correct indentation

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    </CreateAccountRequest>
+                </t:request>
+                <t:response>
+                    <t:select path="//admin:CreateAccountResponse/admin:account" attr="id"  set="test_account1.id"/>
+                    <t:select path='//admin:CreateAccountResponse/admin:account/admin:a[@n="zimbraMailHost"]' set="test_acct.server"/>
+                </t:response>
+            </t:test>
+        </t:test_case>
+        <t:property name="server.zimbraAccount" value="${test_acct.server}"/>
+        <t:test_case testcaseid="acctLogin1_send_message" type="always" >
```
Comment (chinmay15) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:51]: add bugids to all test_case

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                        xmlns="urn:zimbraAdmin">
+                        <name>${admin.user}</name>
+                        <password>${admin.password}</password>
+                    </AuthRequest>
+                </t:request>
+                <t:response>
+                    <t:select path="//admin:AuthResponse/admin:authToken" set="authToken"/>
+                </t:response>
+            </t:test>
+            <t:property name="server.zimbraAccount" value="${test_acct1.server}"/>
```
Comment (chinmay15) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:104]: why is this needed?

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                <t:request>
+                    <CreateAccountRequest
+                        xmlns="urn:zimbraAdmin">
+                        <name>${test_account1.name}</name>
+                        <password>${test_account1.password}</password>
+                    </CreateAccountRequest>
+                </t:request>
+                <t:response>
+                    <t:select path="//admin:CreateAccountResponse/admin:account" attr="id"  set="test_account1.id"/>
+                    <t:select path='//admin:CreateAccountResponse/admin:account/admin:a[@n="zimbraMailHost"]' set="test_acct.server"/>
```
Comment (prashantajari) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:46]: Please remove all unused properties like test_acct1.server, test_acct1.server, server.zimbraAccount. Also, remove setting these properties in CreateAccountResponse

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        <t:test id="ping" required="true">
+            <t:request>
+                <PingRequest
+                    xmlns="urn:zimbraAdmin"/>
+                </t:request>
+                <t:response>
+                    <t:select path="//admin:PingResponse"/>
+                </t:response>
+            </t:test>
+        </t:test_case>
```
Comment (prashantajari) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:21]: do test_case indentation

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                <t:request>
+                    <AuthRequest
+                        xmlns="urn:zimbraAccount">
+                        <account by="name">${test_account1.name}</account>
+                        <password>${test_account1.password}</password>
+                        <!--<prefs/>-->
+                    </AuthRequest>
+                </t:request>
+                <t:response>
+                    <t:select path="//acct:AuthResponse/acct:lifetime"  match="^\d+$"/>
```
Comment (prashantajari) [inline data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml:63]: No need to validate acct:lifetime

**Comment by chinmay15** [comment]
also please attach the generated result text file.

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 5 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: done

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 4 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: done

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 2 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: done

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 6 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: done

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 1 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: done

File: data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml (hunk 3 above)
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/DeleteAccount_WithMsg.xml]: removed it

**Comment by girishbhamare** [comment]
Suggestions from Chinmay and Prashant has been implemented.

**Comment by girishbhamare** [comment]
[DeleteAccount_WithMsg.txt](https://github.com/Zimbra/zm-soap-harness/files/4522455/DeleteAccount_WithMsg.txt)
